### PR TITLE
Remove protected namespace

### PR DESCRIPTION
**Pull request by ygao76**
## Summary
Remove protected namespace and clean up the codes
https://github.com/iron-fish/ironfish/pull/2308
## Testing Plan
unit test 
## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```
@NullSoldier

### CODE REVIEW

File: ironfish/src/sdk.ts [hunk 1]
```diff
@@ -250,7 +242,7 @@ export class IronfishSdk {
             this.config.get('rpcTcpHost'),
             this.config.get('rpcTcpPort'),
             this.logger,
-            namespaces,
+            ALL_API_NAMESPACES,
```
Comment (hughy) [inline ironfish/src/sdk.ts:245]: Auth is still disabled for the RpcTcpAdapter. Perhaps we should get rid of the adapter altogether?

I think it's a question of whether we want to allow users the option to use TCP without encryption or without authentication

**Review by hughy — COMMENTED** [review]
It would be nice to get rid of this 'secure' flag!

File: ironfish/src/sdk.ts (hunk 1 above)
Comment (hairtail) [reply ironfish/src/sdk.ts:245]: I think there are two kind rpc request, public rpc request and private rpc request. For the first one like getBlock, authentication is unnecessary. But our auth system running now checks authentication for all request over tls, but no for request over tcp. And with this PR, all request can be performed over tcp without authentication. My idea in https://github.com/iron-fish/ironfish/pull/2308, use tcp for public request and tls for public + private request. All requests need authentication == no authentication?

**Comment by NullSoldier** [comment]
This PR needs to be rebased it has other peoples commits in it.

**Comment by NullSoldier** [comment]
I changed this PR to run against master now.

File: ironfish/src/sdk.test.ts [hunk 2]
```diff
@@ -152,27 +151,6 @@ describe('IronfishSdk', () => {
...
-      )
-
-      const allowedNamespaces = ALL_API_NAMESPACES.filter(
-        (namespace) => !API_NAMESPACES_PROTECTED.includes(namespace),
-      )
-
-      expect(tcp?.namespaces.sort()).toMatchObject(allowedNamespaces.sort())
-    })
-
     it('should use all RPC namespaces for TCP with rpcTcpSecure flag', async () => {
```
Comment (NullSoldier) [inline ironfish/src/sdk.test.ts:154]: This test should be looked it, it refers to a deleted flag.

File: ironfish/src/sdk.ts (hunk 1 above)
Comment (NullSoldier) [reply ironfish/src/sdk.ts:245]: Got it, you're talking about mounting custom name spaces per adapter. I think that's an interesting concept that we can do but it's outside of this PR.  One thing is that we shouldn't really pre-choose for the user what is mounted and what isn't for usability. My ideas are this.

1. Only mount IPC adapter by default ever
2. Let users enable a public adapter explicitly ```--rpc.tls or --rpc.tcp```
3. Let users manually configure what routes are enabled for each adapter 
   - `ironfish rpc:mount tls node/getStatus`
   - `ironfish rpc:unmount tls node/getStatus`
   - Store this configuration in a file in the data directory

File: ironfish/src/sdk.test.ts (hunk 2 above)
Comment (ygao76) [reply ironfish/src/sdk.test.ts:154]: changed the name

File: ironfish/src/sdk.ts (hunk 1 above)
Comment (hairtail) [reply ironfish/src/sdk.ts:245]: Thanks for your reply. @NullSoldier 
This is complicated. My ideas are this. 
1. All namespace are mounted.
2. Namespace like, node and chain are public, and no authentication needed for this kind of request.
3. Namespace like, account and transaction are private, and authentication needed for this kind of request.
4. Maybe we should refactor our namespace to make all methods of public namespace really public.

Comment (NullSoldier) [reply ironfish/src/sdk.ts:245]: I think there are a lot of instances were nodes were attacked and drained by unsuspecting node runners who didn't realize certain API's were public. I think we should lock everything 100% down if it's public to be safe. Having a public RPC is something that BTC frowned on and removed entirely, read the security section here. https://github.com/bitcoin/bitcoin/blob/master/doc/JSON-RPC-interface.md#security

So we should assume the worst, and only let absolute power users that assume the risk enable these API's publically. Attackers are 100% going to port scan and start making requests against your node in an attempt to DDOS it for sure.

**Review by NullSoldier — APPROVED** [review]

**Pull request merged**